content/maintainers-guide/how-to-become-a-committer.md

```diff
@@ -0,0 +1,61 @@
+    - Display a record of several accepted, quality Pull Requests having received little or no objections from maintainers.
+    - Display a record of Pull Request reviews.
+    - Be active in the project’s [Slack][slack] chat.
+    - Real interest in getting involved.
+2. **Demonstrate a good level of technical knowledge**
+    - Justify a good level of technical skills (SOLID, readable, well-documented, stable, testable code) – in discussions and in code contributions.
+    - Display a good understanding of the project, its architecture, its ecosystem, its constraints (transparency, license, backwards compatibility, dependencies, expected behavior...)
+    - Be competent with git and the GitHub workflow (use a fork, be able to rebase, be able to fetch and update a remote branch...)
+3. **Cultural convergence**
+    - Respect the Code of Conduct; behave respectfully and in good faith.
+    - Agree to work together with the rest of the maintainer team towards main goals (see [The Future Architecture][future-architecture]): migration to Symfony, adoption of VueJS ...
+
+## Requirements to continue being a committer
+
+All the above, plus:
+
+1. **Continuous participation and involvement**
+    - Actively and responsibly review Pull Requests.
+2. **Responsible behavior**
+    - Respect & enforce the project’s quality standards and contribution guidelines.
+    - Respect & enforce the project’s issue and code review workflow.
+    - Respect & enforce convergence with the project’s goals.
+    - Put the best interest of the project before one’s own (in case of conflict of interests).
+
+## How to apply to be a committer
+
+1. [Open an issue on the open source repository](https://github.com/PrestaShop/open-source/issues/new) explaining why you apply to become a committer, and on what part of the project (Core or module/subproject).
+2. Discuss with current maintainers on this issue; particularly, this will be the time for maintainers to analyze the candidate's motivation and skills on the different parts of the project that they applied for.
+3. Once all questions have been answered, current maintainers vote.
+4. If the simple majority of positive votes is achieved, the new committer is approved.
+
+Upon approval, new committer is granted full committer rights and accesses for a 90-day trial period. This period aims to provide both the maintainer team and the new committer a better understanding of this new arrival and to validate whether expectations meet reality. Both maintainer team or new committer can express their wish for this situation to end before the 90-day trial period, revoking the committer status.
+
+At the end of the 90-day trial period, the committer is confirmed in his position.
+
+## How to lose the committer status
+
+A project maintainer can be revoked for any of the following reasons:
+
+- Refusal to follow the rules and policies stated in this document.
+- Lack of activity for the past three months.
```

Comment:
Please define the lack of activity. It's not clear.

One, Ten, Twenty reviews in three months is a lack of activity ?

Reply:
I dont think we should put a number here. You cannot measure the activity of a developer by a raw number of reviews. It would be like trying to measure a developer output by the number of lines of code he/she produces.

You can do twenty useless code reviews on 1 day or you can provide 1 deep code review for one PR with high level feedback.

I know you like numbers and rules @Progi1984 but you wont be able to create and define _relevant and consistent_ rules that apply to real world 😄 . The world and human beings are too messy to be able to fit into tight words.

So just like for Code Review is done by a humain brain and not an automated tool, I suggest "lack of activity" is something that should be discussed and evaluated together as maintainer on a case-by-case basis.

If tomorrow for example the maintainer group thinks I, matks, present a lack of activity, it is a topic to be discussed by maintainers. Then if no consensus "yes or no" can be found, then a vote might be done to decide whether or not I present a lack of activity. Only brains can decide whether there is or not a lack of activity, you cant say "lack of activity is less than X reviews per month". Multiple factors have to be acknowledged. Trust the people, not a random metric.

I think we dont clarify every single word. Actually PrestaShop is an one open source project with the _most detailed process to become maintainer_.  We're already way above average. Look at Symfony, Doctrine, Laravel, PHP, jQuery, VueJS, I was not able to find clear and detailed "how to become a maintainer" guidelines for these projects. When you browse the topics about maintainer what you can see is basically "want to become a maintainer? contribute then we'll chat" 😄  . If I look at open source projects of size and maturity comparable to us, we're doing way better.

If you have an example of three other open source projects of same size / maturity that PrestaShop that have defined "lack of activity for a maintainer is less than X code reviews per month" please share 😄

Reply:
If we can't explain what is lack of activity, we should remove it.

```suggestion
```

Reply:
Hello ! This weekend I found another example about a "broad" maintainer process: https://make.wordpress.org/core/2020/02/09/what-does-it-mean-to-be-a-component-maintainer-a-refresher/

This is a project that we can relate to in terms of size and maturity (although I think they are actually even better than us) and as you can see they have not written rules for everything. The "how to become a maintainer" list is simply "Contact us". No statement about requirements, no statement about "what can make you lose the status"

When I read this I really think we are over-engineering this process. I think we can stay with this "Lack of activity" sentence and say that when we think there is indeed an issue we discuss it, the full maintainer group.

But that is my opinion and I respect that you have another opinion.

This PR has been open for 2 months now and I believe most maintainers agree to it so I merge the PR but we can continue discussing, with the group, this topic and see what the group thinks about this line.
-> https://github.com/PrestaShop/open-source/issues/95

Reply:
I resolve the conversation in the scope of this issue but I'll be happy to discuss this matter with the whole group in https://github.com/PrestaShop/open-source/issues/95